examples/plot_intelex.py

```diff
@@ -0,0 +1,276 @@
+"""
+Creating models that are accelerated by Intel(R) Extension for scikit-learn
+--------------------------------------------------------------------------------
+
+Introduction
+============
+
+This guide demonstrates how Intel(R) Extension for Scikit-learn (also
+``scikit-learn-intelex``, or ``sklearnex``) can be used to speed up inference of
+Scikit-learn models.
+
+The extension supports most of Scikit-learn's classical machine learning
+algorithms, like k-nearest neighbors, support vector machines, linear/logistic
+regression, and more. Stock Scikit-learn implementations are used where no
+optimized version is available, making this package 100% compatible with
+existing code. An up-to-date list of supported algorithms can be found in the
+`official documentation
+<https://intel.github.io/scikit-learn-intelex/algorithms.html>`_.
+
+Intel(R) Extension for Scikit-learn accelerates Scikit-learn algorithms by using
+the latest hardware features and optimized caching and threading strategies.
+Find more details in Intel's blog posts on Medium
+(`1 <https://medium.com/intel-analytics-software/save-time-and-money-with-intel-extension-for-scikit-learn-33627425ae4>`_,
+`2 <https://medium.com/intel-analytics-software/accelerate-your-scikit-learn-applications-a06cacf44912>`_).
+In many cases, optimizations translate to hardware from other vendors, albeit
+with smaller performance gains.
+
+For this example, we train two simple
+:class:`sklearn.neighbors.KNeighborsClassifier` instances, one with and one
+without using ``sklearnex``, and compare inference times. Afterward, we upload
+both models to the Hugging Face Model Hub, download them again, and verify the
+optimization survived being saved in the Hub. Hugging Face Hub supports
+``sklearnex``-optimized models, meaning the achieved speedup will translate for
+Inference API users.
+"""
+
+# %%
+# Imports
+# =======
+# First, we import everything required for the rest of this document.
+
+import os
+import pickle
+from pathlib import Path
+from tempfile import mkdtemp
+from time import perf_counter
+from uuid import uuid4
+
+from huggingface_hub import delete_repo, whoami
+from sklearn.datasets import make_classification
+from sklearn.metrics import log_loss
+from sklearn.model_selection import train_test_split
+from sklearnex import patch_sklearn
+
+from skops import card, hub_utils
+
+# %%
+# Data
+# ====
+# Next, we create some generic data. A dataset of 50k rows x 15 columns is big enough
+# to showcase a performance gain from using ``sklearnex``. Generally speaking,
+# larger datasets will benefit more from the ``sklearnex`` optimizations. More
+# details can be found in the official
+# `README <https://github.com/intel/scikit-learn-intelex/blob/master/README.md>`_.
+X, y = make_classification(
+    n_samples=50_000,
```

Comment:
we'd need to see how long it takes on the CI machines, might need to reduce the number if it's too slow.

Reply:
On my machine the bottleneck is up- and downloading the models. I'm optimistic all the code will run in <10s.